packages/gatsby/src/commands/develop.js

```diff
@@ -350,6 +352,19 @@ module.exports = async (program: any) => {
     console.log()
   }
 
+  function printDeprecationWarnings() {
+    const files = glob
+      .sync("{,!(node_modules|public)/**/}*.js")
```

Comment:
This would be pretty expensive. I was thinking that instead, we could wrap the functions we pass to `boundActionCreators` so that any time those functions are called, we write to console (only once per session) a message saying that they should switch to actions.

Reply:
It is nice you say which files are using it — we do know which plugins we're calling so we could probably capture that data as well to improve the warning.

Reply:
Ah ok, I thought this was searching all of `node_modules`. This is only alerting on site specific code not 3rd party plugins. Makes sense.